src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -0,0 +1,195 @@
+     */
+    public EditEventCommand(Index index, EditEventDescriptor editEventDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editEventDescriptor);
+
+        this.index = index;
+        this.editEventDescriptor = new EditEventDescriptor(editEventDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> currentEventList = model.getFilteredEventList();
+
+        int zeroBased = index.getZeroBased();
+
+        if (zeroBased >= currentEventList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        }
+
+        Event eventToEdit = currentEventList.get(zeroBased);
+        Event editedEvent = createEditedEvent(eventToEdit, editEventDescriptor);
+
+        if (!eventToEdit.isSameEvent(editedEvent) && model.hasEvent(editedEvent)) {
+            throw new CommandException(MESSAGE_DUPLICATE_EVENT);
+        }
+
+        model.setEvent(eventToEdit, editedEvent);
+        model.updateFilteredEventList(Model.PREDICATE_SHOW_ALL_EVENTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_EVENT_SUCCESS, editedEvent));
+    }
+
+    /**
+     * Creates and returns a {@code Event} with the details of {@code eventToEdit} that is edited
+     * with details of {@code editEventDescriptor}
+     */
+    private static Event createEditedEvent(Event eventToEdit, EditEventDescriptor editEventDescriptor) {
+        assert eventToEdit != null;
+
+        String updatedEventTitle = editEventDescriptor.getEventTitle().orElse(eventToEdit.getEventTitle());
+        StartDate updatedDate = editEventDescriptor.getDate().orElse(eventToEdit.getStartDate());
+        StartTime updatedTime = editEventDescriptor.getTime().orElse(eventToEdit.getStartTime());
+        String updatedPurpose = editEventDescriptor.getPurpose().orElse(eventToEdit.getPurpose());
+        return new Event(updatedEventTitle, updatedDate, updatedTime, updatedPurpose);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof EditEventCommand)) {
+            return false;
+        }
+
+        EditEventCommand otherCommand = (EditEventCommand) other;
+        return index.equals(otherCommand.index)
+                && editEventDescriptor.equals(((EditEventCommand) other).editEventDescriptor);
+    }
+
+    /**
+     * Static class that stores the details to edit the event with.
+     * Each field that is non-null will replace the corresponding field value of the event.
+     */
+    public static class EditEventDescriptor {
+        private String eventTitle;
+        private StartDate date;
+        private StartTime time;
+        private String purpose;
+
+        public EditEventDescriptor() {}
+
+        /**
+         * Constructor to create a {@code EditEventDescriptor} object
```

Comment:
May require full stop.
```suggestion
         * Constructor to create a {@code EditEventDescriptor} object.
```